avm/res/aad/domain-service/main.bicep

```diff
@@ -0,0 +1,361 @@
+      "key2": "value2"
+  }
+  '''
+})
+@description('Optional. Tags of the resource.')
+param tags object?
+
+@description('Optional. The lock settings of the service.')
+param lock lockType
+
+@description('Optional. Array of role assignments to create.')
+param roleAssignments roleAssignmentType
+
+@description('Generated. Built-in roles "General": https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles#general.')
+var builtInRoleNames = {
+  Contributor: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'b24988ac-6180-42a0-ab88-20f7382dd24c')
+  Owner: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '8e3af657-a8ff-443c-a75c-2fe8c4bcb635')
+  Reader: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'acdd72a7-3385-48ef-bd42-f606fba81ae7')
+  'Role Based Access Control Administrator': subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'f58310d9-a9f6-439a-9e8d-f62e7b41a168')
+  'User Access Administrator': subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '18d7d88d-d35e-4fb5-a5c3-7773c20a72d9')
+}
+
+// ============== //
+// Resources      //
+// ============== //
+
+resource avmTelemetry 'Microsoft.Resources/deployments@2023-07-01' = if (enableTelemetry) {
+  name: '46d3xbcp.res.aad-domainservice.${replace('-..--..-', '.', '-')}.${substring(uniqueString(deployment().name, location), 0, 4)}'
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+      outputs: {
+        telemetry: {
+          type: 'String'
+          value: 'For more information, see https://aka.ms/avm/TelemetryInfo'
+        }
+      }
+    }
+  }
+}
+
+resource domainservice 'Microsoft.AAD/domainServices@2022-12-01' = {
+  name: name
+  location: location
+  tags: tags
+  properties: {
+    domainName: domainName
+    domainConfigurationType: domainConfigurationType
+    filteredSync: filteredSync
+    notificationSettings: {
+      additionalRecipients: additionalRecipients
+      notifyDcAdmins: notifyDcAdmins
+      notifyGlobalAdmins: notifyGlobalAdmins
+    }
+    ldapsSettings: {
+      externalAccess: externalAccess
+      ldaps: ldaps
+      pfxCertificate: !empty(pfxCertificate) ? pfxCertificate : null
+      pfxCertificatePassword: !empty(pfxCertificatePassword) ? pfxCertificatePassword : null
+    }
+    replicaSets: replicaSets
+    domainSecuritySettings: {
+      tlsV1: tlsV1
+      ntlmV1: ntlmV1
+      syncNtlmPasswords: syncNtlmPasswords
+      syncOnPremPasswords: syncOnPremPasswords
+      kerberosRc4Encryption: kerberosRc4Encryption
+      kerberosArmoring: kerberosArmoring
+    }
+    sku: sku
+  }
+}
+
+resource domainservice_diagnosticSettings 'Microsoft.Insights/diagnosticSettings@2021-05-01-preview' = [for (diagnosticSetting, index) in (diagnosticSettings ?? []): {
```

Comment:
Not using the latest schema. Please update as per [specs](https://azure.github.io/Azure-Verified-Modules/specs/shared/interfaces/#diagnostic-settings), esepecially the 'logs'.